pkg/mtr/mtr.go

```diff
@@ -190,4 +190,5 @@ func (m *MTR) discover(ch chan struct{}, count int) {
 			unknownHopsCount = 0
 		}
 	}
+	close(ch)
```

Comment:
I don't see any returns here, so this should be the same as calling `close(ch)` from `Run`.